src/main/java/bitflow4j/steps/BatchPipelineStep.java

```diff
@@ -189,7 +189,11 @@ private void flush(List<Sample> window) throws IOException {
     // Printing ========================================
     // =================================================
     public String toString() {
-        return String.format("Batch processing, %s handler(s)", handlers.size());
+        String firstHandler = "";
+        if(handlers.size() >= 1){
+            firstHandler = handlers.get(0).getClass().getSimpleName();
+        }
+        return String.format("Batch processing, %s handler(s): %s", handlers.size(), firstHandler);
```

Comment:
IMO, it is misleading to output "handlers: ..." and then only print the first handler class. This implies that all handlers are of the given type, when there is more than one handler. Either print all handlers, or change the message that it says that actually only the first handler is printed.

Reply:
I guess the first handler is enough to recognize the section of the underlying bitflow-script while analyzing log files or do you think the complete list of handlers should be logged here?

Reply:
IMO, one batch processing handler is as important as a full processing step... And since all processing steps are logged, so should be all handlers. It would be even better to print the actual batch handler objects (which could have a better description than just the class), and print one per line. Similar to the output of the Go tool:

```
INFO[Sep  4 15:56:22.495] Pipeline
INFO[Sep  4 15:56:22.495] ├─TCP source on :9999
INFO[Sep  4 15:56:22.495] └─BatchProcessor (3 steps)
INFO[Sep  4 15:56:22.495]   ├─convex hull
INFO[Sep  4 15:56:22.495]   ├─FFT (cut symmetric part: true, normalize: true, fill zeros: false, auto sample frequency, frequency appended at index 0)
INFO[Sep  4 15:56:22.495]   └─Compute & project PCA (0.99 variance)
```

But just a flat list of classes would be a good start :)

Reply:
The log mentioned in this piece of code are printed whenever a batch is flushed. When having multiple batch-steps in a bitflow-script it was not visible which of those batch-steps produced the flush-message. Now all the batch steps involved are being flushed^^